File: patch_links.py
import re

links = {
    ('Catch of the day', 'Presentations'): 'https://canva.link/ii59j786jppq3r3',
    ('Huzons', 'Presentations'): 'https://canva.link/n8yezxjjoqvxsfd',
    ('Mpowered', 'Presentations'): 'https://canva.link/ciueyvrl3lfvter',
    ('Tiktok', 'Presentations'): 'https://canva.link/vridhwunfghwzzh',
    ('Global Food Security', 'Presentations'): 'https://canva.link/vnnn3ea3k20d7hc',
    ('Kommute', 'Presentations'): 'https://canva.link/4xwuge615gxud1i',
    ('Nexu', 'Presentations'): 'https://canva.link/2h4hkdpkmq7xs07',
    ('Sunlife', 'Presentations'): 'https://canva.link/kp0c5ohfxcl0kx1',
    ('Kalan', 'Posters'): 'https://canva.link/kalanphmenu2026',
    ('Iloilo 360', 'Posters'): 'https://drive.google.com/drive/folders/1C95ZR9hXJsSg93flrBDJIzDLsr_WQS8n?usp=sharing',
    ('Jump n Slide', 'Posters'): 'https://drive.google.com/drive/folders/1vuQasJTRLCQE-ukjnlgWx8obAzCPKIxu?usp=sharing',
    ('BMAP', 'Stickers'): 'https://drive.google.com/drive/folders/1ZCH2xBbKqfReHVliMRRVe_79WNjidcsM?usp=sharing',
    ('Iloilo Supermart', 'Video Commercials'): 'https://www.facebook.com/iloilosupermart/videos/to-the-one-who-has-always-been-there-thank-you-mom-for-your-endless-love-guidanc/1403423417516387/',
    ('Kumon', 'Video Commercials'): 'https://www.facebook.com/jeheyms.argel/videos/adfest20-3rd-place-best-tv-ad-3rd-place-best-director3rd-place-best-taglineon-be/1761365211469371/',
    ('Rex x Schoology', 'Video Commercials'): 'https://drive.google.com/file/d/1qQptiCfzXVvq4QO9R8FI1iMeINWghWSt/view?usp=sharing',
    ('BMAP Shirt', 'TShirt'): 'https://drive.google.com/file/d/1nW_AbyipCZQw111Fza9N00vwAGNoPTLZ/view?usp=sharing',
    ('BMAP', 'Social Media Assets'): 'https://www.canva.com/design/DAHPKOPNZNQ/9XsrNA6fNJ8KT-7GmGHqBg/view?utm_content=DAHPKOPNZNQ&utm_campaign=designshare&utm_medium=link2&utm_source=uniquelinks&utlId=hffdf40b191',
    ('Komersyolympics', 'Social Media Assets'): 'https://drive.google.com/drive/folders/15NjSruSOj72OF2BdO_hbCOKY5EHEPK1d?usp=sharing',
    ('Sunlife', 'Social Media Assets'): 'https://drive.google.com/drive/folders/10syooEey1IZvt8Kv9MbFyI_PFXyUyYBm?usp=sharing',
    ('Mpowered', 'Social Media Assets'): 'https://canva.link/rnhs3x1qd9fbcxm',
    ('Uptown Puff', 'Social Media Assets'): 'https://drive.google.com/drive/folders/1WjsSo88p3TB2pB8yJ7wNAqRIzLXlLNkt?usp=sharing'
}

with open('src/pages/WorksPage.tsx', 'r') as f:
    content = f.read()

# The objects start with "    { " and end with "    },"
# Let's split by "    {\n" and "    { "
pattern = r'(?:\s*\{\s*)(.*?)(?:\s*\},|\s*\}\s*])'
# Actually, since it's formatted like:
#    { 
#      id: 23, 
#      title: 'BMAP', 
#      category: 'Social Media Assets', ...
# We can just iterate through each dictionary manually or parse it roughly.

out = []
blocks = re.split(r'(    \{\s*\n|    \{ id: \d+)', content)

# I will just write a simpler matcher:
new_content = content
for (title, cat), link in links.items():
    # we need to find the specific block where title is 'title' and category is 'cat'
    # we can just use regex substitution with a lambda
    pass

import json
# I'll just use the AST approach with a small python parser

File: patch_links.js
const fs = require('fs');

const links = [
  { match: (p) => p.title === 'Catch of the day' && p.category === 'Presentations', link: 'https://canva.link/ii59j786jppq3r3' },
  { match: (p) => p.title === 'Huzons' && p.category === 'Presentations', link: 'https://canva.link/n8yezxjjoqvxsfd' },
  { match: (p) => p.title === 'Mpowered' && p.category === 'Presentations', link: 'https://canva.link/ciueyvrl3lfvter' },
  { match: (p) => p.title === 'Tiktok' && p.category === 'Presentations', link: 'https://canva.link/vridhwunfghwzzh' },
  { match: (p) => p.title === 'Global Food Security' && p.category === 'Presentations', link: 'https://canva.link/vnnn3ea3k20d7hc' },
  { match: (p) => p.title === 'Kommute' && p.category === 'Presentations', link: 'https://canva.link/4xwuge615gxud1i' },
  { match: (p) => p.title === 'Nexu' && p.category === 'Presentations', link: 'https://canva.link/2h4hkdpkmq7xs07' },
  { match: (p) => p.title === 'Sunlife' && p.category === 'Presentations', link: 'https://canva.link/kp0c5ohfxcl0kx1' },
  { match: (p) => p.title === 'Kalan' && p.category === 'Posters', link: 'https://canva.link/kalanphmenu2026' },
  { match: (p) => p.title === 'Iloilo 360' && p.category === 'Posters', link: 'https://drive.google.com/drive/folders/1C95ZR9hXJsSg93flrBDJIzDLsr_WQS8n?usp=sharing' },
  { match: (p) => p.title === 'Jump n Slide' && p.category === 'Posters', link: 'https://drive.google.com/drive/folders/1vuQasJTRLCQE-ukjnlgWx8obAzCPKIxu?usp=sharing' },
  { match: (p) => p.title === 'BMAP' && p.category === 'Stickers', link: 'https://drive.google.com/drive/folders/1ZCH2xBbKqfReHVliMRRVe_79WNjidcsM?usp=sharing' },
  { match: (p) => p.title === 'Iloilo Supermart' && p.category === 'Video Commercials', link: 'https://www.facebook.com/iloilosupermart/videos/to-the-one-who-has-always-been-there-thank-you-mom-for-your-endless-love-guidanc/1403423417516387/' },
  { match: (p) => p.title === 'Kumon' && p.category === 'Video Commercials', link: 'https://www.facebook.com/jeheyms.argel/videos/adfest20-3rd-place-best-tv-ad-3rd-place-best-director3rd-place-best-taglineon-be/1761365211469371/' },
  { match: (p) => p.title === 'Rex x Schoology' && p.category === 'Video Commercials', link: 'https://drive.google.com/file/d/1qQptiCfzXVvq4QO9R8FI1iMeINWghWSt/view?usp=sharing' },
  { match: (p) => p.title === 'BMAP Shirt' && p.category === 'TShirt', link: 'https://drive.google.com/file/d/1nW_AbyipCZQw111Fza9N00vwAGNoPTLZ/view?usp=sharing' },
  { match: (p) => p.title === 'BMAP' && p.category === 'Social Media Assets', link: 'https://www.canva.com/design/DAHPKOPNZNQ/9XsrNA6fNJ8KT-7GmGHqBg/view?utm_content=DAHPKOPNZNQ&utm_campaign=designshare&utm_medium=link2&utm_source=uniquelinks&utlId=hffdf40b191' },
  { match: (p) => p.title === 'Komersyolympics' && p.category === 'Social Media Assets', link: 'https://drive.google.com/drive/folders/15NjSruSOj72OF2BdO_hbCOKY5EHEPK1d?usp=sharing' },
  { match: (p) => p.title === 'Sunlife' && p.category === 'Social Media Assets', link: 'https://drive.google.com/drive/folders/10syooEey1IZvt8Kv9MbFyI_PFXyUyYBm?usp=sharing' },
  { match: (p) => p.title === 'Mpowered' && p.category === 'Social Media Assets', link: 'https://canva.link/rnhs3x1qd9fbcxm' },
  { match: (p) => p.title === 'Uptown Puff' && p.category === 'Social Media Assets', link: 'https://drive.google.com/drive/folders/1WjsSo88p3TB2pB8yJ7wNAqRIzLXlLNkt?usp=sharing' }
];

let content = fs.readFileSync('src/pages/WorksPage.tsx', 'utf8');

// Find the start and end of allProjects array
const startIndex = content.indexOf('const allProjects = [');
if (startIndex !== -1) {
    // I need to properly parse it or use regex. 
    // Since objects are well-structured, I can use regex to inject externalLink
}
